rebing/graphql-laravel
# [CHORES] Attend to phpstan, tests and other things currently failing

## Summary
- Fix newly reported phpstan issues, adapt baseline
- tests related
  - Removed outdated application version checks in tests
  - add a custom `assertMatchesRegularExpression` which provides it for our older PHPUnit versions and gets us rid of the deprecation warning for now
  - Replace deprecated `at()` matcher using `withConsecutive`
  - workaround change in orchestra/testbench-core which changes validation messages (due to a change in Laraval)
    => see also https://github.com/orchestral/testbench-core/commit/6c9c77b2e978890cb6a2712251ddab5eb1b79049
- Github Actions related
  - disable fail-fast: our matrix is so varied, it helps to run them all always
  - fix scheduled tests being run on Monday (and not Sunday)
- Code style related:
  - Get rid of matt-allan/laravel-code-style and pure php-cs-fixer with `@PSR12`
    This package always created subtle dependencies constraints on _our_ Laravel dependencies because it itself does depend on illuminate/support.
    Also: the Laravel code style is not formalize specified (except their use of StyleCI), there's no reason to stick to it anyway -> we'll just go with PSR-12 and might add more rules later if we feel like it
- phpstan related
  - Bump nunomaduro/larastan to 0.7.0
  - Refine some use and add new checks

---

Type of change:
- [x] Misc. change (internal, infrastructure, maintenance, etc.)

Checklist:
- [ ] Existing tests have been adapted and/or new tests have been added
- [ ] Add a CHANGELOG.md entry
- [ ] Update the README.md
- [x] Code style has been fixed via `composer fix-style`
